target/ppc/translate/vsx-impl.c.inc

```diff
@@ -1659,6 +1659,96 @@ static bool trans_XXSPLTI32DX(DisasContext* ctx, arg_8RR_D_IX *a)
     return true;
 }
 
+static bool trans_LXVKQ(DisasContext *ctx, arg_X_uim5 *a)
+{
+    TCGv tmp;
+
+    REQUIRE_INSNS_FLAGS2(ctx, ISA310);
+    REQUIRE_VSX(ctx);
+
+    switch (a->uim) {
```

Comment:
Might be simpler with a lookup table:
```
static const uint64_t values[] = {
    0, /* unspecified */
    0x3FFF000000000000ULL, /* QP +1.0 */
    0x4000000000000000ULL, /* QP +2.0 */
(...)
};
```

Even with an ``if`` to check for invalid values of uim:
```
if (a->uim == 0 || (a->uim > 0b1001 && a->uim < 0b10000) || a->uim >= ARRAY_SIZE(values)) {
    gen_invalid(ctx);
} else {
    set_cpu_vsrl(a->xt, tcg_constant_i64(0));
    set_cpu_vsrh(a->xt, tcg_constant_i64(values[a->uim]));
}
return true;
```

A few notes:
- I added the 'ULL' suffix to the 64-bit unsigned constants
- For unspecified values of uim, instead of returning false, I'm generating an invalid exception and returning true (even tough I don't know if the ISA mandates that this should be considered invalid). And I'm returning true because if we get to this situation, we should consider that the instruction was handled
- I used ``tcg_constant_i64`` instead of ``tcg_constant_tl``, because it must always be 64-bits, right?

Reply:
Adding a lookup table, we might also add a valid mask: https://github.com/PPC64/qemu/pull/20/commits/e0387ee13307b161c62535bc0a2e4a124b8d662f
The suffix and `tcg_constant_i64` were fixed in https://github.com/PPC64/qemu/pull/20/commits/406b6a8ad6c048d20ab89876ed3ba2911a963c12